flow360/component/simulation/services.py

```diff
@@ -120,26 +135,28 @@ def get_default_params(
     unit_system = init_unit_system(unit_system_name)
     dummy_value = 0.1
     with unit_system:
-        params = SimulationParams(
-            reference_geometry=ReferenceGeometry(
-                area=1, moment_center=(0, 0, 0), moment_length=(1, 1, 1)
-            ),
-            meshing=MeshingParams(
-                volume_zones=[AutomatedFarfield(name="Farfield")],
-            ),
-            operating_condition=AerospaceCondition(velocity_magnitude=dummy_value),
+        reference_geometry = ReferenceGeometry(
+            area=1, moment_center=(0, 0, 0), moment_length=(1, 1, 1)
         )
+        operating_condition = AerospaceCondition(velocity_magnitude=dummy_value)
 
-    if length_unit is not None:
-        # Store the length unit so downstream services/pipelines can use it
-        # pylint: disable=fixme
-        # TODO: client does not call this. We need to start using new webAPI for that
-        with _model_attribute_unlock(params.private_attribute_asset_cache, "project_length_unit"):
-            # pylint: disable=assigning-non-slot,no-member
-            params.private_attribute_asset_cache.project_length_unit = LengthType.validate(
-                length_unit
-            )
     if root_item_type == "Geometry":
+        automated_farfield = AutomatedFarfield(name="Farfield")
+        with unit_system:
+            params = SimulationParams(
+                reference_geometry=reference_geometry,
+                meshing=MeshingParams(
```

Comment:
Why initialize unit_system context twice here? In particular, should the farfield be initialized outside this context? This looks like a change compared to how it was before..

Reply:
No, `unit_system` context is initialised one time each, there is no nested `unit_system` context. SimulationParams are always in one unit context and stores it. Other objects do not store unit system but only apply units.